src/world/common/DeadUnkNpcAIMainFunc5.inc.c

```diff
@@ -0,0 +1,143 @@
+            npc->flags &= ~NPC_FLAG_ENABLE_HIT_SCRIPT;
+        } else {
+            npc->flags &= ~NPC_FLAG_GRAVITY;
+            npc->flags |= NPC_FLAG_ENABLE_HIT_SCRIPT;
+        }
+
+        if (enemy->unk_B0 & 4) {
+            s32 emoteTemp;
+
+            script->functionTemp[0] = 99;
+            script->functionTemp[1] = 0;
+            fx_emote(2, npc, 0.0f, npc->collisionHeight, 1.0f, 2.0f, -20.0f, 0x28, &emoteTemp);
+            enemy->unk_B0 &= ~4;
+        } else if (enemy->flags & ENEMY_FLAGS_40000000) {
+            script->functionTemp[0] = 12;
+            enemy->flags &= ~ENEMY_FLAGS_40000000;
+        }
+    }
+
+    if (enemy->varTable[9] > 0) {
+        enemy->varTable[9]--;
+        if (enemy->varTable[9] == 0) {
+            if (npc->currentAnim.w == 0x55002E || npc->currentAnim.w == 0x55002F) {
+                npc->currentAnim.w = 0x55000C;
+            }
+        } else {
+            return ApiStatus_BLOCK;
+        }
+    }
+
+    switch (script->functionTemp[0]) {
+        case 0:
+            func_800495A0(script, aiSettings, territoryPtr);
+            npc->collisionHeight = enemy->varTable[6];
+        case 1:
+            func_800496B8(script, aiSettings, territoryPtr);
+            break;
+        case 2:
+            base_UnkNpcAIFunc1(script, aiSettings, territoryPtr);
+            if (enemy->varTable[7] == 6) {
+                if (rand_int(100) < 33) {
+                    if (enemy->varTable[8] != 0) {
+                        enemy->varTable[8] = 0;
+                        enemy->unk_B5 = 0;
+                        npc->currentAnim.w = 0x55002F;
+                    } else {
+                        enemy->varTable[8] = 1;
+                        enemy->unk_B5 = 1;
+                        npc->currentAnim.w = 0x55002E;
+                    }
+                    enemy->varTable[9] = 7;
+                    return ApiStatus_BLOCK;
+                }
+            }
+        case 3:
+            func_80049C04(script, aiSettings, territoryPtr);
+            break;
+        case 12:
+            N(set_script_owner_npc_anim)(script, aiSettings, territoryPtr);
+        case 13:
+            N(UnkDistFunc)(script, aiSettings, territoryPtr);
+            break;
+        case 14:
+            N(UnkNpcAIFunc12)(script, aiSettings, territoryPtr);
+            break;
+        case 15:
+            N(set_script_owner_npc_col_height)(script, aiSettings, territoryPtr);
+            break;
+        case 99:
+            func_8004A73C(script);
+            break;
+    }
+
+    if (enemy->varTable[7] == 6) {
+        if (enemy->varTable[8] != 0) {
+            enemy->unk_B5 = 1;
+        } else {
+            enemy->unk_B5 = 0;
+        }
+        if (enemy->varTable[8] != 0) {
+            switch (npc->currentAnim.w - 0x550004) {
+                case 0:
+                case 8:
+                case 10:
+                case 12:
+                case 14:
+                case 18:
+                case 20:
```

Comment:
This switch should remove the - 0x550004 and carry it into each case, using the defined values for anims instead.